backend/DB/api/routes.py

```diff
@@ -0,0 +1,239 @@
+from backend.DB.api import app, db, queries
+from flask import jsonify, request, abort, render_template
+
+from backend.DB.api.tables import User, Pair, Reward, Threshold
+
+
+@app.route('/')
+def graph_display():
+    return render_template('graph_display.html')
+
+
+def format_users(users):
+    output = []
+    for user in users:
+        output.append(
+            {'username': user.username, 'name': user.name, 'image': user.image, 'active': user.active})
+    return jsonify(output)
+
+
+@app.route('/status/data')
+def status_data():
+    return jsonify({"cake_count": len(queries.get_pair_since_last_reward("cake")),
+                     "pizza_count": len(queries.get_pair_since_last_reward("pizza")),
+                     "pizza_thres": queries.get_threshold("pizza").threshold,
+                     "cake_thres": queries.get_threshold("cake").threshold,
+                     "unused_cake": queries.get_unused_rewards_count_by_type("cake"),
+                     "unused_pizza": queries.get_unused_rewards_count_by_type("pizza"),
+                     "last_pair": ["esog", "mleik"]})
+
+
+@app.route('/api/user/add', methods=['POST'])
+def add_user():
+    r = request.get_json()
+
+    if r is None or 'username' not in r or 'name' not in r:
+        abort(400)
+
+    username = r['username']
+    name = r['name']
+    user = User(username, name)
+    queries.add_user(user)
+    return format_users([user])
+
+
+@app.route('/api/user/all', methods=['GET'])
+def get_all_users():
+    users = queries.get_all_users()
+    return format_users(users)
+
+
+@app.route('/api/user/active', methods=['GET'])
+def get_active_users():
+    users = queries.get_active_users()
+    return format_users(users)
+
+
+@app.route('/api/user/get/<username>', methods=['GET'])
+def get_user(username):
+    user = queries.get_user_by_username(username)
+    return format_users([user])
+
+
+@app.route('/api/user/update', methods=['PUT'])
+def update_user():
+    r = request.get_json()
+
+    if r is None or 'username' not in r or 'name' not in r or 'active' not in r:
```

Comment:
This was a little tricky to read. The json should possibly be parsed here to make the check more robust